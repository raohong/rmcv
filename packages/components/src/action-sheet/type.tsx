export type ActionSheetAction = {
  /**
   * @description 标题
   */
  name: React.ReactNode;
  /**
   * @description 二级标题
   */
  subName?: React.ReactNode;
  /**
   * @description 文字颜色
   */
  color?: string;
  /**
   * @description 选项自定义 class
   */
  className?: string;
  /**
   * @description 是否为加载状态
   */
  loading?: boolean;
  /**
   * @description 是否为禁用状态
   */
  disabled?: boolean;
  /**
   * @description 点击时触发的回调函数
   */
  callback?: (action: ActionSheetAction) => void;
};

export type ActionSheetProps = {
  /**
   * @description 是否显示动作面板
   */
  visible?: boolean;
  /**
   * @description 面板选项列表
   */
  actions?: ActionSheetAction[];
  /**
   * @description 动作面板标题
   */
  title?: React.ReactNode;
  /**
   * @description 动作面板描述
   */
  description?: React.ReactNode;
  /**
   * @description 关闭按钮文字
   */
  cancelText?: React.ReactNode;
  /**
   * @description 是否可关闭的
   * @default true
   */
  closable?: boolean;
  /**
   * @description 关闭图标
   */
  closeIcon?: React.ReactNode;
  /**
   * @description 是否显示圆角
   * @default true
   */
  round?: boolean;
  /**
   * @description 自定义 class
   */
  className?: string;
  /**
   * @description 是否显示 overlay
   * @default true
   */
  overlay?: boolean;
  /**
   * @description overlay 自定义 className
   */
  overlayClassName?: string;
  /**
   * @description overlay 自定义 style
   */
  overlayStyle?: React.CSSProperties;
  /**
   * @description overlay 点击是否关闭
   * @default true
   */
  overlayClosable?: boolean;
  /**
   * @description 是否在显示弹层时才渲染节点
   * @default true
   */
  lazyRender?: boolean;
  /**
   * @description
   * @default true
   */
  lockScroll?: boolean;
  /**
   * @description 是否在页面回退时自动关闭
   */
  closeOnPopState?: boolean;
  /**
   * @description 是否开启底部安全区适配
   */
  safeArea?: boolean;
  /**
   * @description 点击选项后是否关闭
   */
  closeOnClickAction?: boolean;
  /**
   * @description 点击选项时触发，禁用或加载状态下不会触发
   */
  onSelect?: (action: ActionSheetAction, index: number) => void;
  /**
   * @description 关闭回调
   */
  onClose?: () => void;
  /**
   * @description 点击取消按钮，此时还要触发 onClose
   */
  onCancel?: () => void;
  /**
   * @description 点击 overlay 回调
   */
  onOverlayClick?: () => void;
  /**
   * @description 关闭前的回调函数，返回 false 可阻止关闭，支持返回 Promise
   */
  onBeforeClose?: (
    action: ActionSheetAction,
  ) => void | boolean | Promise<boolean>;
  /**
   * @description 动画结束后回调
   */
  afterVisibleChange?: (visible: boolean) => void;
  /**
   * @description 自定义内容
   */
  children?: React.ReactNode;
};
