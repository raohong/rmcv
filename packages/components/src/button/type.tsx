import type { JSXIntrinsicElementProps } from '../types';
import type { LoadingProps } from '../loading';

export type ButtonSize = 'large' | 'default' | 'small' | 'mini';

export type ButtonType = 'primary' | 'success' | 'warning' | 'danger' | 'link';

export type ButtonShape = 'round' | 'square';

export type NativeButtonHTMLType = 'button' | 'submit' | 'reset';

type BaseButtonProps = {
  /**
   * @description 按钮类型
   */
  type?: ButtonType;
  /**
   * @description 按钮尺寸
   */
  size?: ButtonSize;
  /**
   * @description 是否是块级元素
   */
  block?: boolean;
  /**
   * @description 是否镂空
   */
  plain?: boolean;
  /**
   * @description 按钮形状
   */
  shape?: ButtonShape;
  /**
   * @description 按钮 disabled
   */
  disabled?: boolean;
  /**
   * @description 按钮 icon
   */
  icon?: React.ReactElement;
  /**
   * @description 是否 loading
   */
  loading?: boolean;
  /**
   * @description border 是否是高清
   */
  hairline?: boolean;
  /**
   * @@description 点击事件
   */
  onClick?: React.MouseEventHandler<HTMLElement>;
  /**
   * @@description 按钮内容
   */
  children?: React.ReactNode;
  /**
   * @@description loading 图标文字
   */
  loadingText?: string;
  /**
   * @@description loading 图标类型
   */
  loadingType?: LoadingProps['type'];
  /**
   * @@description loading 图标大小
   * @default 20px
   */
  loadingSize?: LoadingProps['size'];
};

type AnchorButtonProps = {
  /**
   * @description 按钮 html 链接
   */
  href?: string;
  target?: string;
} & BaseButtonProps &
  JSXIntrinsicElementProps<'button', 'onClick' | 'type'>;

type NativeButtonProps = {
  /**
   * 按钮的原始 type
   */
  htmlType?: NativeButtonHTMLType;
} & BaseButtonProps &
  JSXIntrinsicElementProps<'button', 'onClick' | 'type'>;

export type ButtonProps = Partial<NativeButtonProps & AnchorButtonProps>;
