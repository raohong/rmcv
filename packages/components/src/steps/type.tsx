import React from 'react';

export type StepsDirection = 'horizontal' | 'vertical';

export type StepsStatus = 'active' | 'inactive' | 'finish';

export type StepsProps = {
  /**
   * @description 当前步骤，从0开始计数
   */
  current?: number;
  /**
   * @description 当前步骤变换时回调
   */
  onChange?: (current?: number) => void;
  /**
   * @description 布局方向
   * @default vertical
   */
  direction?: StepsDirection;
  /**
   * @description 当前步骤的图标
   */
  activeIcon?: React.ReactNode;
  /**
   * @description 非当前步骤的图标
   */
  inactiveIcon?: React.ReactNode;
  /**
   * @description 已完成步骤的图标，优先级大于 inactiveIcon
   */
  finishIcon?: React.ReactNode;
  /**
   * @description 当前步骤和已完成步骤的颜色
   */
  activeColor?: string;
  /**
   * @description 未激活步骤的颜色
   */
  inactiveColor?: string;
} & Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'>;

export type StepProps = {
  /**
   * @description 步骤条内容
   */
  children?: React.ReactNode;
  /**
   * @description 当前步骤的图标
   */
  icon?: React.ReactNode;
  /**
   * @description 当前步骤状态
   */
  status?: StepsStatus;
} & React.HTMLAttributes<HTMLDivElement>;
