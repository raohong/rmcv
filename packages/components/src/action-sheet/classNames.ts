import { generateComponentClassNameUtility } from '@rmc-vant/system';
import {
  ActionSheetComponentState,
  ActionSheetNSlot,
  ActionSheetSlot,
} from './interface';

export const ActionSheetName = 'ActionSheet';

export const {
  actionSheetClassNames,
  composeActionSheetSlotClassNames,
  getActionSheetSlotClassNames,
  getActionSheetClassName,
} = generateComponentClassNameUtility<
  typeof ActionSheetName,
  ActionSheetComponentState,
  ActionSheetSlot,
  ActionSheetNSlot
>(
  ActionSheetName,
  {
    root: true,
    title: true,
    description: true,
    content: true,
    cancelButton: true,
    item: true,
    itemSubname: true,
    loadingIcon: true,
    closeIcon: true,
    itemDisabled: true,
    itemLoading: true,
  },
  () => ({
    root: ['root'],
    title: ['title'],
    closeIcon: ['closeIcon'],
    description: ['description'],
    content: ['content'],
    cancelButton: ['cancelButton'],
    item: ['item'],
    itemSubname: ['itemSubname'],
    loadingIcon: ['loadingIcon'],
  }),
);
