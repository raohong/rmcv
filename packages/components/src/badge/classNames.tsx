import { generateComponentClassNameUtility } from '@rmc-vant/system';
import { camelCase } from '@rmc-vant/utils';
import type { BadgeComponentState, BadgeNSlot, BadgeSlot } from './interface';

export const BadgeName = 'Badge';

export const {
  badgeClassNames,
  getBadgeSlotClassNames,
  composeBadgeSlotClassNames,
} = generateComponentClassNameUtility<
  typeof BadgeName,
  BadgeComponentState,
  BadgeSlot,
  BadgeNSlot
>(
  BadgeName,
  {
    root: true,
    number: true,
    numberWrapper: true,
    wrapper: true,
    dot: true,
    positionBottomLeft: true,
    positionBottomRight: true,
    positionTopLeft: true,
    positionTopRight: true,
  },
  ({ dot, position }) => ({
    root: ['root', camelCase(`position-${position}`)],
    wrapper: ['wrapper', dot && 'dot'],
    number: ['number'],
    numberWrapper: ['numberWrapper'],
  }),
);
