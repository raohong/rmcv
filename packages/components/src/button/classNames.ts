import { generateComponentClassNameUtility } from '@rmc-vant/system';
import { camelCase } from '@rmc-vant/utils';
import type { ButtonComponentState, ButtonNSlot, ButtonSlot } from './interface';

export const ButtonName = 'Button';

export const {
  buttonClassNames,
  getButtonSlotClassNames,
  composeButtonSlotClassNames,
} = generateComponentClassNameUtility<
  typeof ButtonName,
  ButtonComponentState,
  ButtonSlot,
  ButtonNSlot
>(
  ButtonName,
  {
    root: true,
    icon: true,
    loading: true,
    disabled: true,
    outlined: true,
    outlinedSizeMini: true,
    outlinedSizeSmall: true,
    outlinedSizeLarge: true,
    outlinedPrimary: true,
    outlinedSuccess: true,
    outlinedWarning: true,
    outlinedDanger: true,
    outlinedShapeRound: true,
    outlinedShapeSquare: true,
    contained: true,
    containedSizeMini: true,
    containedSizeSmall: true,
    containedSizeLarge: true,
    containedPrimary: true,
    containedSuccess: true,
    containedWarning: true,
    containedDanger: true,
    containedShapeRound: true,
    containedShapeSquare: true,
    loadingIcon: true,
    block: true,
  },
  ({ variant, size, shape, loading, disabled, type, block }) => ({
    root: [
      'root',
      variant,
      loading && 'loading',
      disabled && 'disabled',
      size !== 'normal' && camelCase(`${variant}-size-${size}`),
      type !== 'default' && camelCase(`${variant}-${type}`),
      shape !== 'default' && camelCase(`${variant}-shape-${shape}`),
      block && 'block',
    ],
    icon: ['icon'],
    loadingIcon: ['loadingIcon'],
  }),
);
