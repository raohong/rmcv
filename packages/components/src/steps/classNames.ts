import { generateComponentClassNameUtility } from '@rmc-vant/system';
import { camelCase } from '@rmc-vant/utils';
import type {
  StepComponentState,
  StepNSlot,
  StepSlot,
  StepsComponentState,
  StepsNSlot,
  StepsSlot,
} from './interface';

export const StepsName = 'Steps';

export const StepName = 'Step';

export const {
  stepsClassNames,
  composeStepsSlotClassNames,
  getStepsSlotClassNames,
} = generateComponentClassNameUtility<
  typeof StepsName,
  StepsComponentState,
  StepsSlot,
  StepsNSlot
>(
  StepsName,
  {
    root: true,

    vertical: true,
    horizontal: true,
  },
  ({ direction }) => ({
    root: ['root', direction],
  }),
);

export const { stepClassNames, composeStepSlotClassNames, getStepSlotClassNames } =
  generateComponentClassNameUtility<
    typeof StepName,
    StepComponentState,
    StepSlot,
    StepNSlot
  >(
    StepName,
    {
      root: true,
      title: true,
      tail: true,
      icon: true,
      iconWrapper: true,
      dot: true,
      process: true,
      processDot: true,
      processIcon: true,
      processTail: true,
      processTitle: true,
      finish: true,
      finishDot: true,
      finishIcon: true,
      finishTail: true,
      finishTitle: true,
      wait: true,
      waitDot: true,
      waitIcon: true,
      waitTail: true,
      waitTitle: true,
    },
    ({ status }) => ({
      root: ['root', status],
      title: ['title', camelCase(`${status}-title`)],
      tail: ['tail', camelCase(`${status}-tail`)],
      icon: ['icon', camelCase(`${status}-icon`)],
      dot: ['dot', camelCase(`${status}-dot`)],
      iconWrapper: ['iconWrapper'],
    }),
  );
