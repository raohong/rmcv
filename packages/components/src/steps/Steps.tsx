import { useControllableValue } from '@rmc-vant/hooks';
import { CheckedFilled } from '@rmc-vant/icons';
import { useComponentTheme } from '@rmc-vant/system';
import { isNil, isNumber, isString, omit } from '@rmc-vant/utils';
import clsx from 'clsx';
import React, { useMemo } from 'react';
import { useThemeProps } from '../config-provider';
import Step from './Step';
import { StepsName, composeStepsSlotClassNames } from './classNames';
import type { StepStatus, StepsComponentState, StepsProps } from './interface';
import { StepsRoot } from './styles';

const Steps = React.forwardRef<HTMLDivElement, StepsProps>((props, ref) => {
  const {
    activeIcon,
    inactiveIcon,
    finishIcon,
    activeColor,
    inactiveColor,
    className,
    items,
    direction = 'horizontal',
    ...rest
  } = useThemeProps(StepsName, props);
  const [current, setCurrent] = useControllableValue(props, {
    defaultValue: 0,
    valuePropName: 'current',
  });
  const { palette } = useComponentTheme();
  const componentState = useMemo<StepsComponentState>(
    () => ({
      direction,
      inactiveColor: inactiveColor ?? palette.gray600,
      activeColor: activeColor ?? palette.primary,
    }),
    [direction, activeColor, inactiveColor, palette],
  );

  const slotClassNames = composeStepsSlotClassNames(componentState);

  const getStepStatus = (index: number): StepStatus => {
    if (index < current) {
      return 'finish';
    }

    if (index === current) {
      return 'process';
    }

    return 'wait';
  };

  const getIcon = (status: StepStatus) => {
    if (status === 'process') {
      return activeIcon ?? <CheckedFilled />;
    }

    if (status === 'finish') {
      return finishIcon ?? inactiveIcon;
    }

    return inactiveIcon;
  };

  const clickable = 'onChange' in props;

  return (
    <StepsRoot
      className={clsx(className, slotClassNames.root)}
      ref={ref}
      {...omit(rest, ['current', 'onChange'])}
      componentState={componentState}
    >
      {items?.map((item, index) => {
        const status = getStepStatus(index);
        const content = item.children ?? item.label;
        const key
          = isNil(item.key) && (isString(content) || isNumber(content))
            ? content
            : item.key;

        return (
          <Step
            key={key}
            onClick={() => {
              if (clickable) {
                setCurrent(index);
              }
            }}
            status={status}
            icon={getIcon(status)}
            stepsComponentState={componentState}
            {...omit(item, ['label'])}
            children={content}
          />
        );
      })}
    </StepsRoot>
  );
});

export default Steps;
