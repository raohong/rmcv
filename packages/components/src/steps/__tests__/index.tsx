import { render, screen } from '@testing-library/react';
import React from 'react';
import Steps from '../Steps';
import { stepClassNames } from '../classNames';
import type { StepsProps } from '../interface';

const testId = 'steps';

const App: React.FC<StepsProps> = props => (
  <Steps
    data-testid={testId}
    items={[
      { 'label': 1, 'data-testid': 's1' },
      { 'label': 2, 'data-testid': 's2' },
      { 'label': 3, 'data-testid': 's3' },
    ]}
    {...props}
  />
);

it('render correctly', () => {
  const tree = render(<App />);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with current', () => {
  render(<App current={1} />);

  expect(screen.getByTestId('s1')).toHaveClass(stepClassNames.finish);
  expect(screen.getByTestId('s2')).toHaveClass(stepClassNames.process);
});

it('render with onChange', () => {
  const onChange = jest.fn();

  render(<App onChange={onChange} />);

  const list = screen
    .getByTestId(testId)
    .querySelectorAll(`.${stepClassNames.root}`);

  list.forEach((item) => {
    expect(item).toHaveAttribute('role', 'button');
  });
  screen.getByTestId('s2').click();
  expect(onChange).toHaveBeenCalledWith(1);
});

it('render with actionIcon', () => {
  render(<App activeIcon='activeIcon' />);

  expect(
    screen.getByTestId('s1').querySelector(`.${stepClassNames.icon}`),
  ).toContainHTML('activeIcon');
});

it('render with inactionIcon', () => {
  render(<App inactiveIcon='inactiveIcon' current={1} />);

  expect(
    screen.getByTestId('s1').querySelector(`.${stepClassNames.icon}`),
  ).toContainHTML('inactiveIcon');
  expect(
    screen.getByTestId('s3').querySelector(`.${stepClassNames.icon}`),
  ).toContainHTML('inactiveIcon');
});

it('render with inactionIcon and finishICon', () => {
  render(<App inactiveIcon='inactiveIcon' finishIcon='finishIcon' current={1} />);

  expect(
    screen.getByTestId('s1').querySelector(`.${stepClassNames.icon}`),
  ).toContainHTML('finishIcon');
  expect(
    screen.getByTestId('s3').querySelector(`.${stepClassNames.icon}`),
  ).toContainHTML('inactiveIcon');
});
