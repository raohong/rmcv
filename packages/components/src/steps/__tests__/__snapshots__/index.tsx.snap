// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  background: #fff;
  padding: 22px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: inherit;
  border: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
  cursor: not-allowed;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  padding: 0 0 22px 0;
  line-height: 1;
}

.emotion-1:first-of-type .RVuiStep-iconWrapper {
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-1:not(:first-of-type):not(:last-child) .RVuiStep-title {
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-1:last-child {
  position: absolute;
  right: 10px;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  text-align: right;
}

.emotion-1:last-child .RVuiStep-iconWrapper {
  right: calc(8px / -2);
  left: unset;
  -webkit-transform: translate(0, -50%);
  -moz-transform: translate(0, -50%);
  -ms-transform: translate(0, -50%);
  transform: translate(0, -50%);
}

.emotion-2 {
  font-size: 12px;
  color: #1989fa;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1.4285714285714286;
}

.emotion-3 {
  position: absolute;
  top: 30px;
  left: 0;
  -webkit-transform: translateY(-50%) scaleY(0.5);
  -moz-transform: translateY(-50%) scaleY(0.5);
  -ms-transform: translateY(-50%) scaleY(0.5);
  transform: translateY(-50%) scaleY(0.5);
  transform-origin: bottom center;
  width: 100%;
  height: 1px;
  background-color: #ebedf0;
}

.emotion-3:last-child {
  display: none;
}

.emotion-4 {
  position: absolute;
  top: 30px;
  left: -8px;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  background-color: #fff;
  padding: 0 8px;
  z-index: 1;
  line-height: 0;
}

.emotion-5 {
  font-size: 12px;
  color: #1989fa;
}

.emotion-6 {
  display: inline-block;
  color: inherit;
  font-style: normal;
  line-height: 0;
  box-sizing: border-box;
  text-rendering: optimizeLegibility;
  -webkit-font-smooth: antialiased;
}

.emotion-6 * {
  line-height: 1;
}

.emotion-8 {
  font-size: 12px;
  color: #969799;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  line-height: 1.4285714285714286;
}

.emotion-11 {
  display: block;
  width: 5px;
  height: 5px;
  border-radius: 50%;
  background: #969799;
}

<div
    class="RVuiSteps-root RVuiSteps-horizontal emotion-0"
    data-testid="steps"
  >
    <div
      class="RVuiTouchable-root RVuiStep-root RVuiStep-process emotion-1"
      data-testid="s1"
      role="button"
      type="button"
    >
      <div
        class="RVuiStep-title RVuiStep-processTitle emotion-2"
      >
        1
      </div>
      <div
        class="RVuiStep-tail RVuiStep-processTail emotion-3"
      />
      <div
        class="RVuiStep-iconWrapper emotion-4"
      >
        <span
          class="RVuiStep-icon RVuiStep-processIcon emotion-5"
        >
          <span
            aria-label="Checked-Filled"
            class="RVuiIcon-root emotion-6"
            role="img"
          >
            <svg
              fill="currentColor"
              height="1em"
              version="1.1"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512.45568 57.344c251.35104 0 455.11168 203.76064 455.11168 455.11168 0 251.35104-203.76064 455.11168-455.11168 455.11168C261.10464 967.56736 57.344 763.80672 57.344 512.45568 57.344 261.10464 261.10464 57.344 512.45568 57.344z m272.444416 288.876544c-5.842944-5.445632-16.21504-5.953536-22.995968-1.074176l-0.315392 0.232448-301.232128 228.097024c-5.515264 4.176896-15.601664 4.72064-21.655552 1.186816l-0.320512-0.193536-113.998848-71.015424c-7.031808-4.380672-16.93696-2.89792-22.390784 3.219456l-0.24576 0.2816-13.169664 15.497216c-5.553152 6.53312-4.62848 15.625216 1.519616 21.586944l0.27136 0.258048 140.93312 131.342336c10.247168 9.550848 26.529792 10.051584 36.615168 0.9728l0.313344-0.28672 323.310592-301.310976c6.679552-6.223872 7.225344-15.606784 1.036288-21.63712l-0.254976-0.241664-7.419904-6.915072z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
    <div
      class="RVuiTouchable-root RVuiStep-root RVuiStep-wait emotion-1"
      data-testid="s2"
      role="button"
      type="button"
    >
      <div
        class="RVuiStep-title RVuiStep-waitTitle emotion-8"
      >
        2
      </div>
      <div
        class="RVuiStep-tail RVuiStep-waitTail emotion-3"
      />
      <div
        class="RVuiStep-iconWrapper emotion-4"
      >
        <div
          class="RVuiStep-dot RVuiStep-waitDot emotion-11"
        />
      </div>
    </div>
    <div
      class="RVuiTouchable-root RVuiStep-root RVuiStep-wait emotion-1"
      data-testid="s3"
      role="button"
      type="button"
    >
      <div
        class="RVuiStep-title RVuiStep-waitTitle emotion-8"
      >
        3
      </div>
      <div
        class="RVuiStep-tail RVuiStep-waitTail emotion-3"
      />
      <div
        class="RVuiStep-iconWrapper emotion-4"
      >
        <div
          class="RVuiStep-dot RVuiStep-waitDot emotion-11"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
